src/mlpack/tests/mean_shift_test.cpp

```diff
@@ -110,32 +110,34 @@ TEMPLATE_TEST_CASE("MeanShiftSimpleCentroidsOnlyTest", "[MeanShiftTest]", float,
 TEMPLATE_TEST_CASE("GaussianClustering", "[MeanShiftTest]", float, double)
 {
   typedef TestType ElemType;
+  typedef typename arma::Mat<ElemType> MatType;
 
-  GaussianDistribution g1("0.0 0.0 0.0", arma::eye<arma::mat>(3, 3));
-  GaussianDistribution g2("5.0 5.0 5.0", 2 * arma::eye<arma::mat>(3, 3));
-  GaussianDistribution g3("-3.0 3.0 -1.0", arma::eye<arma::mat>(3, 3));
-  GaussianDistribution g4("6.0 -2.0 -2.0", 3 * arma::eye<arma::mat>(3, 3));
+  GaussianDistribution<MatType> g1("0.0 0.0 0.0", arma::eye<MatType>(3, 3));
+  GaussianDistribution<MatType> g2("5.0 5.0 5.0", 2 * arma::eye<MatType>(3, 3));
+  GaussianDistribution<MatType> g3("-3.0 3.0 -1.0", arma::eye<MatType>(3, 3));
+  GaussianDistribution<MatType> g4("6.0 -2.0 -2.0",
+      3 * arma::eye<MatType>(3, 3));
 
   // We may need to run this multiple times, because sometimes it may converge
   // to the wrong number of clusters.
   bool success = false;
   for (size_t trial = 0; trial < 4; ++trial)
   {
-    arma::Mat<ElemType> dataset(3, 4000);
+    MatType dataset(3, 4000);
     for (size_t i = 0; i < 1000; ++i)
-      dataset.col(i) = arma::conv_to<arma::Col<ElemType>>::from(g1.Random());
+      dataset.col(i) = g1.Random();
     for (size_t i = 1000; i < 2000; ++i)
-      dataset.col(i) = arma::conv_to<arma::Col<ElemType>>::from(g2.Random());
+      dataset.col(i) = g2.Random();
     for (size_t i = 2000; i < 3000; ++i)
-      dataset.col(i) = arma::conv_to<arma::Col<ElemType>>::from(g3.Random());
+      dataset.col(i) = g3.Random();
     for (size_t i = 3000; i < 4000; ++i)
-      dataset.col(i) = arma::conv_to<arma::Col<ElemType>>::from(g4.Random());
+      dataset.col(i) = g4.Random();
```

Comment:
I have seen these in several places, how are we able to get rid of the conv_to  as we are assigning mat to col?